File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    var answer = [];
    var days = [];

    for (let i = 0; i < progresses.length; i++) {
        const remaining = Math.ceil((100 - progresses[i]) / speeds[i]);
        days.push(remaining);
    }
    
    while (days.length > 0) {
        const current = days.shift();
        let cnt = 1;
        
        while (days.length > 0 && days[0] <= current) {
            days.shift();
            cnt++;
        }
        
        answer.push(cnt);
    }
    
    return answer;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
from collections import deque

def solution(progresses, speeds):
    q = deque([])
    answer = []
    
    for i in range(len(progresses)):
        remaining = (100 - progresses[i]) / speeds[i]
        q.append(int(remaining) if remaining.is_integer() else int(remaining) + 1)
    
    while q:
        current = q.popleft()
        cnt = 1
        
        while q and q[0] <= current:
            q.popleft()
            cnt += 1
        
        answer.append(cnt)
    
    return answer
